tools/FileTestDefs.sh

```diff
@@ -518,10 +549,32 @@ prevent_cpp_includes_test() {
 }
 standard_checks+=(prevent_cpp_includes)
 
+# Prevent editing RunSingleTest files. We may need to occasionally update these
+# files, but it's at least less than once a year.
+prevent_run_single_test_changes() {
+    [[ $1 =~ "RunSingleTest/CMakeLists.txt" ]]
+}
+prevent_run_single_test_changes_report() {
+    echo "Found changes to RunSingleTest/CMakeLists.txt"
+    pretty_grep "include .*\.cpp" "$@"
+}
+prevent_run_single_test_changes_test() {
+    test_check pass ./tests/Unit/RunSingleTest/Test.cpp ''
+    test_check pass src/Domain/Blah.cpp ''
+    test_check pass ./src/Domain/Blah.cpp ''
+    test_check fail ./tests/Unit/RunSingleTest/CMakeLists.txt ''
+    test_check fail RunSingleTest/CMakeLists.txt ''
+    test_check fail ../RunSingleTest/CMakeLists.txt ''
+}
+
 # if test is enabled: redefines staged_grep to run tests on files that
 # are not in git
-[ "$1" = --test ] && staged_grep() { grep "$@"; } && \
-    run_tests "${standard_checks[@]}"
+if [ "$1" = --test ] ; then
+    checks_for_test=${standard_checks}
+    checks_for_test+=(prevent_run_single_test_changes)
```

Comment:
I don't think this works.  (From experiment, I think you lose all but the first standard check.)  A simpler way to do this is
```bash
[ "$1" = --test ] && staged_grep() { grep "$@"; } && \
￼    run_tests "${standard_checks[@]}" prevent_run_single_test_changes
```

Reply:
Ah thanks! I tried that initially with `"${standard_checks[@]} prevent..."`. I was going to ask you about this :)